src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -595,14 +565,15 @@ public bool TryCopyTo(scoped TensorSpan<T> destination)
         {
             bool retVal = false;
 
-            if (_flattenedLength <= destination.FlattenedLength)
+            if (_shape.FlattenedLength <= destination.FlattenedLength)
             {
                 scoped Span<nint> curIndexes;
                 nint[]? curIndexesArray;
-                if (Rank > 6)
+                if (Rank > TensorSpan.MaxRankForStackAlloc)
```

Comment:
nit: I think `MaxRankForStackAlloc` is the wrong name and `TensorSpan` may be the wrong place.

I think instead this should be something like `TensorShape.MaxInlineRank` or something to that affect